src/storage/mod.rs

```diff
@@ -21,18 +21,22 @@ use std::io::Error as IoError;
 use std::sync::{atomic, Arc};
 use std::{cmp, error, u64};
 
-use engine::{IterOption, DATA_KEY_PREFIX_LEN};
+use engine::{
```

Comment:
I saw it in the [import formatting](https://github.com/nrc/pingcap-style-guide/blob/master/src/rust/formatting.md#imports):
> Group imports from the same logical project together (i.e., imports from different crates in the same project).
> Group imports from external crates together; do not separate std imports from other crates.

`engine` crate is a component in TiKV. `futures` is an external crate. `kvproto` is a part of TiKV but in a different GitHub repo.  Should we group them together?

Reply:
Oh, do you think they all belong to external crates? Only `crate::..` is from the same logical project.

Reply:
I've been thinking about it since I was doing this PR :-) I think I will change that advice somewhat, it is complicated.

The grouping I;ve used here is to put the imports then the re-exports and group by std, external crates (including workspace crates), then the current crate